tests/rules/method.js

```diff
@@ -494,6 +502,26 @@ eslintTester.run("method", rule, {
                     type: "CallExpression"
                 }
             ]
-        }
+        },
+        { // AssignmentExpression, ensure we are detecting the pattern from the right part - Regression test for #124
+            code: "(e = node.insertAdjacentHTML)('beforebegin', evil)",
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    message: "Unsafe call to node.insertAdjacentHTML for argument 1",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        { // Regression test for #124, make sure we go deeper and detect the unsafe pattern
+            code: "(e = node.insertAdjacentHTML('beforebegin', evil))()",
```

Comment:
Thinking about this, shouldn't this give us a Syntax Tree that, when traversed, looks no different than `node.insertAdjacentHTML('beforebegin', evil)`? I'm not sure how we can test that the test adds something
Would this be possible with coverage? We can do that as a follow-up, I'm happy to merge this as it is.

Reply:
this is now #128